modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockProcessor.scala

```diff
@@ -0,0 +1,104 @@
+package org.tessellation.dag.l1.domain.block
+
+import cats.Applicative
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.Async
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.semigroup._
+import cats.syntax.traverse._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.block.BlockValidator
+import org.tessellation.dag.block.BlockValidator.BlockValidationError
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.dag.l1.domain.address.storage.AddressStorage
+import org.tessellation.dag.l1.domain.block.BlockProcessor.{BalanceNegative, BlockFailedValidation}
+import org.tessellation.dag.l1.domain.transaction.TransactionService
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.schema.transaction.Transaction
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.{Hashed, SecurityProvider}
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import io.estatico.newtype.ops.toCoercibleIdOps
+
+class BlockProcessor[F[_]: Async: KryoSerializer: SecurityProvider](
```

Comment:
btw - can it be simply a BlockService? I think we don't need to create additional services and split the logic between them

Reply:
I will give it some thought. If we move it to `BlockService` then it will require both `TransactionService` and `StorageService` inside, if we further move the balance acceptance to `TransactionService` we will need to have the `AddressStorage` there as well. So it's getting nested and in the end `BlockService` will be kind of doing everything. I would rather go with flatter structure with dependencies between services decreased to minimum. But that's just what I feel comfortable with. I will look into it and see - maybe I'm overdramatic.

Reply:
I am all in favor of having many small services rather than a few bigger ones.

Reply:
Me too, but this is what we had in v1.0 and because of many dependent services we decided to combine them to reduce complexity. And now, we’re doing something opposite 🤔

Reply:
You can always have many smaller ones and use fascade pattern by creating single `BlockService`